src/TVar.cpp

```diff
@@ -72,108 +85,143 @@ TVar::TVar( TVar * p, QString kName, int kt, QString val, int vt )
 {
 }
 
-void TVar::setReference(bool s){
+void TVar::setReference( const bool s )
+{
     reference = s;
 }
 
-void TVar::addChild(TVar * c){
+void TVar::addChild( TVar * c )
+{
     children.append(c);
 }
 
-QString TVar::getName(){
+QString TVar::getName()
+{
     return name;
 }
 
-bool TVarLessThan( TVar* var, TVar* var2 ){
-    QString a = var->getName();
-    QString b = var2->getName();
-    if ( a.toInt() && b.toInt() )
+bool TVarLessThan( TVar * varA, TVar * varB )
+{
+    QString a = varA->getName();
+    QString b = varB->getName();
+    bool isAOk = false;
+    bool isBOk = false;
+
+    // Previously we do not check the result of a toInt() call on the QStrings
+    // but they would happly return a zero value for a QString that can not be
+    // converted to a number and then the IF branch would be taken regardless
+    // of whether one or both of the QStrings was NOT actually a number
+    if( a.toInt( & isAOk ) && b.toInt( & isBOk ) && isAOk && isBOk ) {
         return a.toInt() < b.toInt();
-    return a.toLower() < b.toLower();
+    }
+    else {
+        return a.toLower() < b.toLower();
```

Comment:
In hindsight, looking at the Qt Docs:

> If cs is Qt::CaseSensitive, the comparison is case sensitive; otherwise the comparison is case insensitive.
> Case sensitive comparison is based exclusively on the numeric Unicode values of the characters and is very fast, but is not what a human would expect. Consider sorting user-visible strings with localeAwareCompare().
> 
```
>   int x = QString::compare("aUtO", "AuTo", Qt::CaseInsensitive);  // x == 0
>   int y = QString::compare("auto", "Car", Qt::CaseSensitive);     // y > 0
>   int z = QString::compare("auto", "Car", Qt::CaseInsensitive);   // z < 0

>```

So 


    return a.compare( b, Qt::CaseInsensitive );

would also work;  or possibly, for a more Unicode friendly way of doing it:

    return a.localeAwareCompare( b );

but this may/can not ignore case or perhaps, as according to the Qt Doc's:

> The comparison is performed in a locale- and also platform-dependent manner. Use this function to present sorted lists of strings to the user.
>
> On OS X and iOS this function compares according the "Order for sorted lists" setting in the International preferences panel.
>